app.rb

```diff
@@ -0,0 +1,60 @@
+#$stdout.puts 'clear'
+# $tout.puts "Please select a calculator: (a) or (b)"
+# $stdout.flush
+# selection = $stdin.gets.chomp
+# $stdout.puts "you have selected #{selection}"
+
+def output text
+  $stdout.puts text
+  $stdout.flush
+end
+
+def input
+  return $stdin.gets.chomp
+end
+
+output "hello, would you like to use a (b)asic or (a)dvanced calculator?";
+selection = input
+if selection == "b"
+output "Would you like to use the (a)dd (d)ivide, (m)ultiply or (s)ubtract?";
+basicFunction = input
+elsif selection == "a"
+output "Would you like to use the (s)quare root or (p)ower of?"
+advancedFunction = input
+
```

Comment:
You should have ended this conditional statement here. Because you don't, it means that the next set of conditions will never be checked and your app stops here.